.github/copilot-instructions.md

```diff
@@ -0,0 +1,57 @@
+# Copilot Instructions for CLI for Microsoft 365
+
+## Project Overview
+- This is a cross-platform CLI tool to manage Microsoft 365 tenants and SharePoint Framework (SPFx) projects.
+- Written in TypeScript, targeting Node.js 22+ (see `package.json`, `tsconfig.json`).
+- Main entry: `src/index.ts` → `src/cli/cli.ts` (command parsing/execution).
+- Commands are modular, organized under `src/m365/` by workload (e.g., `spo`, `aad`, `teams`).
+- Each command is a class extending `Command` (`src/Command.ts`).
+- Command metadata and options: see `src/cli/CommandInfo.ts`, `src/cli/CommandOptionInfo.ts`.
+- Command discovery/build: see `scripts/write-all-commands.js` (generates `allCommands.json`).
+
+## Key Patterns & Conventions
+- **Global options**: All commands accept `--output`, `--query`, `--debug`, `--verbose` (see `src/GlobalOptions.ts`).
+- **Authentication**: Managed via `src/Auth.ts`, supports multiple auth types (Device Code, Certificate, Secret, Managed Identity, etc.).
+- **HTTP requests**: Use the wrapper in `src/request.ts` (Axios-based, with logging/debug hooks).
+- **Telemetry**: Centralized in `src/telemetry.ts`.
+- **Validation**: Use Zod schemas (`zod` import, see `src/Command.ts`).
+- **Prompting**: Use utilities in `src/utils/prompt.ts` for interactive input.
+- **Config**: User settings/configuration via `configstore` (see `src/config.ts`).
+- **Completion**: Shell completion logic/scripts in `scripts/Register-CLIM365Completion.ps1` and `scripts/Test-CLIM365Completion.ps1`.
+
+## Developer Workflows
+- **Build**: `npm run build` (TypeScript compile + command metadata generation)
+- **Test**: `npm test` (runs version check, lint, and Mocha tests)
+- **Lint**: `npm run lint` (uses custom ESLint rules from `eslint-rules/`)
+- **Watch**: `npm run watch` (TypeScript in watch mode)
+- **Symlink for local CLI**: `npm link` (after build)
+- **Node version**: Must be 22 (see `scripts/check-version.js`)
```

Comment:
again I wonder if it is a huge benefit to specify the node version. For sure another place for us to remember